apps/aehttp/src/aehttp_helpers.erl

```diff
@@ -720,6 +728,25 @@ dry_run_txs_([#{ <<"tx">> := ETx } | Txs], Acc) ->
 dry_run_txs_([#{ <<"call_req">> := CallReq } | Txs], Acc) ->
     dry_run_txs_(Txs, [{call_req, CallReq} | Acc]).
 
+
+dry_run_tx_hash(TxHash) ->
+    case aeser_api_encoder:safe_decode(tx_hash, TxHash) of
+        {ok, TxHashInternal} ->
+            case aec_chain:find_tx_with_location(TxHashInternal) of
+                {mempool, SignedTx} ->
+                    Tx = aetx_sign:tx(SignedTx),
+                    {Type, _} = aetx:specialize_type(Tx),
+                    case not lists:member(Type, [paying_for_tx, offchain_tx, ga_meta_tx]) of
```

Comment:
At this stage dry-running `paying_for_tx` and `ga_meta_tx` isn't really an issue - but I suspect it might be stopped further down the chain 🤔

Reply:
This is the same check as for a transaction. I should be put them in a function to make sure they are consistent for mempool transactions and contributed transactions.